src/main.py

```diff
@@ -51,66 +52,94 @@ class CameraApp(App):
 
     STREAM_NAMES = ["rgb", "disparity", "left", "right"]
 
-    def __init__(self, service_config: EventServiceConfig, stream_every_n: int) -> None:
+    def __init__(self, service_config: EventServiceConfig) -> None:
         super().__init__()
+
         self.service_config = service_config
-        self.stream_every_n = stream_every_n
 
-        self.image_decoder = ImageDecoder()
-        self.image_subscription_tasks: list[asyncio.Task] = []
+        self.image_decoder = TurboJPEG()
+
+        self.view_name = "rgb"
+
+        self.async_tasks: list[asyncio.Task] = []
 
     def build(self):
         return Builder.load_file("res/main.kv")
 
     def on_exit_btn(self) -> None:
         """Kills the running kivy application."""
+        for task in self.tasks:
+            task.cancel()
         App.get_running_app().stop()
 
+    def update_view(self, view_name: str):
+        self.view_name = view_name
+
     async def app_func(self):
         async def run_wrapper():
             # we don't actually need to set asyncio as the lib because it is
             # the default, but it doesn't hurt to be explicit
             await self.async_run(async_lib="asyncio")
-            for task in self.image_subscription_tasks:
+            for task in self.async_tasks:
                 task.cancel()
 
+        config_list = proto_from_json_file(
+            self.service_config, EventServiceConfigList()
+        )
+
+        oak0_client: EventClient | None = None
+
+        for config in config_list.configs:
+            if config.name == "oak0":
+                oak0_client = EventClient(config)
+
+        if None in [oak0_client]:
```

Comment:
Since only one client
```suggestion
        if oak0_client is None:
```